src/lib/utils/utils.js

```diff
@@ -89,45 +89,93 @@ function getJson(url, cb) {
   httpGetJson(url, cb);
 }
 
-function pingEndpoint(host, port, type, protocol, origin, callback) {
-  const options = {
-    protocolVersion: 13,
-    perMessageDeflate: true,
-    origin: origin,
-    host: host,
-    port: port
-  };
-  if (type === 'ws') {
-    options.headers = {
-      'Sec-WebSocket-Version': 13,
-      Connection: 'Upgrade',
-      Upgrade: 'websocket',
-      'Sec-WebSocket-Extensions': 'permessage-deflate; client_max_window_bits',
-      Origin: origin
-    };
-  }
-  let req;
+function pingEndpoint(host, port, type, protocol, origin, callback, count = 0) {
   // remove trailing api key from infura, ie rinkeby.infura.io/nmY8WtT4QfEwz2S7wTbl
-  if (options.host.indexOf('/') > -1) {
-    options.host = options.host.split('/')[0];
-  }
-  if (protocol === 'https') {
-    req = require('https').get(options);
-  } else {
-    req = require('http').get(options);
-  }
+  const _host = host.indexOf('/') > -1 ? host.split('/')[0] : host;
+  const max_retries = 600;
+  const timeout = 1000;
 
-  req.on('error', (err) => {
-    callback(err);
-  });
+  let alreadyClosed = false;
+  let retrying = false;
 
-  req.on('response', (_response) => {
-    callback();
-  });
+  const doCallback = (v) => {
```

Comment:
What is `v`? Can we have a better name

Reply:
Also, seems like `doCallback` returns a boolean about if it should callback, but the name implies that it will actually callback. We should change that name too to something like `shouldCallback`

Reply:
It was a silent "should" in front of the variable name... You're right, `shouldCallback` is better, will change it.

Reply:
Done.